docs/index.md

```diff
@@ -359,16 +357,41 @@ But this problem might be resolved with `soft` method inside the first parser si
 ```scala mdoc
 val fieldValueSoft = alpha.rep.string.soft ~ pchar(':')
 
-val p4 = fieldValueSoft.? ~ (searchWord ~ sp.?).rep.string
+val p3 = fieldValueSoft.? ~ (searchWord ~ sp.?).rep.string
 
-p4.parse("title:The Wind Has Risen")
+p3.parse("title:The Wind Has Risen")
 // res2 = Right((,(Some((title,())),The Wind Has Risen)))
-p4.parse("The Wind Has Risen")
-// res3 = Right((,(None,The Wind Has Risen)))
+p3.parse("The Wind Has Risen")
+// res3 = Right((,(None,The Wind 22Has Risen)))
 ```
 
 So when the _right side_ returns an epsilon failure the `soft` method allows us to rewind parsed input and try to proceed it's parsing with next parsers (without changing the parser itself!).
 
+Another common use case for `soft` is implementing a parser to find values interspersed with a separator. For example, if you want to extract `'a'`, `'b'` and `'c'` from `"a,b,c"`, you can use `soft`.
+
+Naively, one may try the following:
+
+```scala mdoc
+val p4 = alpha
+val naiveInterspersed = (p4 <* pchar(',')).rep
+naiveInterspersed.parse("a,b,c")
+// res4 = Left(Error(5,NonEmptyList(InRange(offset = 5, lower = ',', upper = ','),List())))
+```
+
+Basically, it's looking for the trailing comma:
+
+```scala mdoc
+naiveInterspersed.parse("a,b,c,")
+// res5 = Right(("", NonEmptyList('a', List('b', 'c'))))
+```
+
+But you can use, `soft` along with `|` to implement that:
+```scala mdoc
+val interspersed = ((p4.soft <* pchar(',')) | p4).rep
```

Comment:
this example would also parse `abc` which maybe what you want but might be a bit confusing for newcomers.

here is an example of how I parse lists in bosatsu:
https://github.com/johnynek/bosatsu/blob/c58b55785b6ac72e59b98afe3149a042d623a902/core/src/main/scala/org/bykn/bosatsu/Parser.scala#L302